enterprise/trackedusers/users_reporter.go

```diff
@@ -0,0 +1,223 @@
+	}
+	workspaceSourceUserIdTypeMap := make(map[string]map[string]map[string]*hll.Hll)
+	for _, job := range jobs {
+		if job.WorkspaceId == "" {
+			u.log.Warn("workspace_id not found in job", logger.NewIntField("jobId", job.JobID))
+			continue
+		}
+
+		sourceID := gjson.GetBytes(job.Parameters, "source_id").String()
+		if sourceID == "" {
+			u.log.Warn("source_id not found in job parameters", obskit.WorkspaceID(job.WorkspaceId),
+				logger.NewIntField("jobId", job.JobID))
+			continue
+		}
+
+		if sourceIDtoFilter != nil && sourceIDtoFilter[sourceID] {
+			u.log.Debug("source to filter", obskit.SourceID(sourceID))
+			continue
+		}
+		userID := gjson.GetBytes(job.EventPayload, "batch.0.userId").String()
+		anonymousID := gjson.GetBytes(job.EventPayload, "batch.0.anonymousId").String()
+
+		if userID == "" && anonymousID == "" {
+			u.log.Warn("both userID and anonymousID not found in job event payload", obskit.WorkspaceID(job.WorkspaceId),
+				logger.NewIntField("jobId", job.JobID))
+			continue
+		}
+
+		if workspaceSourceUserIdTypeMap[job.WorkspaceId] == nil {
+			workspaceSourceUserIdTypeMap[job.WorkspaceId] = make(map[string]map[string]*hll.Hll)
+		}
+
+		if userID != "" {
+			workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID] = u.recordIdentifier(workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID], userID, idTypeUserID)
+		}
+
+		if anonymousID != "" {
+			workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID] = u.recordIdentifier(workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID], anonymousID, idTypeAnonymousID)
+		}
+
+		if userID != "" && anonymousID != "" {
+			combinedUserIDAnonymousID := combineUserIDAnonymousID(userID, anonymousID)
+			workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID] = u.recordIdentifier(workspaceSourceUserIdTypeMap[job.WorkspaceId][sourceID], combinedUserIDAnonymousID, idTypeUserIDAnonymousIDCombination)
+		}
+	}
+
+	if len(workspaceSourceUserIdTypeMap) == 0 {
+		u.log.Warn("no data to collect", obskit.WorkspaceID(jobs[0].WorkspaceId))
+		return nil
+	}
+
+	reports := make([]*UsersReport, 0)
+	for workspaceID, sourceUserMp := range workspaceSourceUserIdTypeMap {
+		sourceIDReports := make([]*UsersReport, 0, len(sourceUserMp))
+		sourceIDReports = append(sourceIDReports, lo.MapToSlice(sourceUserMp, func(sourceID string, userIdTypeMap map[string]*hll.Hll) *UsersReport {
+			return &UsersReport{
+				WorkspaceID:              workspaceID,
+				SourceID:                 sourceID,
+				UserIDHll:                userIdTypeMap[idTypeUserID],
+				AnonymousIDHLL:           userIdTypeMap[idTypeAnonymousID],
+				IdentifiedAnonymousIDHLL: userIdTypeMap[idTypeUserIDAnonymousIDCombination],
+			}
+		})...)
+		reports = append(reports, sourceIDReports...)
+	}
+	return reports
+}
+
+func (u *UniqueUsersReporter) ReportUsers(ctx context.Context, reports []*UsersReport, tx *txn.Tx) error {
+	if len(reports) == 0 {
+		return nil
+	}
+	stmt, err := tx.PrepareContext(ctx, pq.CopyIn("tracked_users_reports",
```

Comment:
Should we move this to a constant?
```suggestion
	stmt, err := tx.PrepareContext(ctx, pq.CopyIn(trackUsersTable,
```